lib/portlayer/logging/logging.go

```diff
@@ -135,11 +135,26 @@ func Join(h interface{}) (interface{}, error) {
 		return nil, fmt.Errorf("Type assertion failed for %#+v", handle)
 	}
 
+	var logFilePath string
+
 	VMPathName := handle.Spec.VMPathName()
 	VMName := handle.Spec.Spec().Name
+	VMID := handle.Spec.ID()
+	dsName := handle.Spec.Datastore.Name()
+
+	dsType, err := handle.Spec.Datastore.Type(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("failed to check datastore type for %s: %s", VMID, err)
+	}
+
+	logFilePath = fmt.Sprintf("[%s] %s", dsName, VMID)
+	if dsType == types.HostFileSystemVolumeFileSystemTypeVsan {
+		logFilePath = fmt.Sprintf("%s/%s", VMPathName, VMName)
+	}
 
 	for _, logFile := range []string{"tether.debug", "output.log"} {
-		filename := fmt.Sprintf("%s/%s/%s", VMPathName, VMName, logFile)
+		filename := fmt.Sprintf("%s/%s", logFilePath, logFile)
```

Comment:
@emlin If it is on a vsan setup, the `VMPathName` would become `[datastore_name]`, whereas what we want here is `[datastore_name]/vmname`